packages/backend/src/handlers/actions/guild/discord/oauthHandler.ts

```diff
@@ -138,18 +147,27 @@ const createNewGuild = async (
   }
 
   const createGuildResponse = await client.CreateGuild({
-    objects: newGuildPayload,
+    object: newGuildPayload,
   });
-  if (
-    createGuildResponse.insert_guild != null &&
-    createGuildResponse.insert_guild.returning.length > 0
-  ) {
-    const newGuild = createGuildResponse.insert_guild?.returning[0];
-    const successResponse: DiscordGuildAuthResponse = {
-      success: true,
-      guildname: newGuild.guildname,
+  if (createGuildResponse.insert_guild_one != null) {
+    const newGuild = createGuildResponse.insert_guild_one;
+
+    const newGuildMetadataPayload: Guild_Metadata_Insert_Input = {
+      creator_id: creatorId,
+      discord_id: discordGuild.id,
+      guild_id: newGuild.id,
+      discord_metadata: discordMetadata,
     };
-    return successResponse;
+    const createGuildMetadataResponse = await client.CreateGuildMetadata({
+      object: newGuildMetadataPayload,
+    });
+    if (createGuildMetadataResponse.insert_guild_metadata_one != null) {
+      const successResponse: DiscordGuildAuthResponse = {
+        success: true,
+        guildname: newGuild.guildname,
+      };
+      return successResponse;
+    }
   }
   throw new Error();
```

Comment:
Would it make sense to throw a more meaningful error? (not sure if this is logged anywhere though)